### ISSUE TITLE
Error when dimension is odd number

### ISSUE THREAD

**Issue by caioartus** [lucidrains/rotary-embedding-torch#45]
Hello ! 
When the dimension passed to RotaryEmbedding() is odd, an error arrises. For example if dim = 9 :  
The expanded size of the tensor (9) must match the existing size (8) at non-singleton dimension 1.  Target sizes: [727, 9].  Tensor sizes: [727, 8]

This did not happen in previous versions (like 0.6.5).  

I've pin pointed it to the dimension of the cached_freqs. Its caused because freqs has shape (dim//2, ) and cached_freqs has shape (max_seq_len, dim). So when dim is not even torch fails to expand in this line `self.cached_freqs[:seq_len] = freqs.detach()` (line 347 of rotary_embedding_torch.py) 

Best regards, 
Caio
